src/minichain/mempool.py

```
@@ -0,0 +1,247 @@
+        return transaction_id in self._entries_by_id
+
+    def add_transaction(
+        self,
+        transaction: Transaction,
+        state: State,
+        *,
+        received_at: int | None = None,
+    ) -> str:
+        """Validate and enqueue a transaction, returning its transaction id."""
+        if transaction.is_coinbase():
+            raise MempoolValidationError("Coinbase transactions are not accepted")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)